### PR TITLE
tgui: Flex basis should be zero if grow is used

### PR DESCRIPTION
**Pull request by stylemistake**
## About The Pull Request

**Potentially breaking change!**

Developers who don't know any better always make the same mistake over and over again: they use `<Flex.Item grow={1}>` without understanding, whether they need to specify the basis (which should be `<Flex.Item grow={1} basis={0}>`.

So to make Flex more approachable to beginners, I am changing Flex to use saner defaults.

**Old behavior**: Regardless of `grow`, basis defaults to `basis="content"`.

This makes the growth ratio unequal and varying, because the final width is affected by the *size of content*.

**New behavior**: When using `grow={1}`, basis defaults to `basis={0}`.

This makes the growth ratio absolute, because basis is set to zero and size of content plays no role in calculation of item widths.

### Rationale behind the change

Grow is most commonly used in its shorthand CSS form:

```css
flex: 1
```

This rule expands into:

```css
flex-grow: 1;
flex-shrink: 0;
flex-basis: 0;
```

### CODE REVIEW

**Comment by stylemistake** [comment]
Heavy flex/stack users: @tralezab @Mothblocks @Timberpoes @Watermelon914

**Comment by Mothblocks** [comment]
I'm going to cut out time tomorrow or the day after to see how prefs menu is affected before asking for a test merge.

**Comment by Timberpoes** [comment]
I was sure I commented here, but maybe I just closed the tab without commenting.

Remembering basis={0} with my grow was something I failed at a lot when first working with Flex-like components. I definitely think it should be the default behaviour and will make things more intuitive. It's definitely worth a test merge to see which (if any) interfaces it breaks, but I feel this is a good QoL change to aspiring tgui devs.

**Review by Watermelon914 — APPROVED** [review]

**Comment by Mothblocks** [comment]
image

This is the only prefs menu regression I've been able to notice, I'll PR a fix to your branch.

File: tgui/packages/tgui/interfaces/PreferencesMenu/QuirksPage.tsx [hunk 1]
```diff
@@ -73,7 +73,7 @@ const QuirkList = (props: {
                     <Stack fill style={{
                       "font-size": "1.2em",
                     }}>
-                      <Stack.Item grow>
+                      <Stack.Item grow basis="content">
```
Comment (stylemistake) [inline tgui/packages/tgui/interfaces/PreferencesMenu/QuirksPage.tsx:76]: But why is it grow even?

Comment (Mothblocks) [reply tgui/packages/tgui/interfaces/PreferencesMenu/QuirksPage.tsx:76]: i dont remember

**Comment by stylemistake** [comment]
Deconflicted

**Comment by Mothblocks** [comment]
Will ask for a TM

**Comment by Watermelon914** [comment]
Bugs out for integrated circuit UI
image

**Comment by MrMelbert** [comment]
i think this PR breaks the changeling antag hud, they cannot read their objectives
image

i think this PR also breaks the medical kiosk UI
image

**Review by Watermelon914 — APPROVED** [review]

**Pull request merged**